src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CopyRatioExpectations.java

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.Arrays;
+
+/**
+ * This class represents copy ratio (or copy number) posterior (or prior) expectations
+ * as reported by implementations of {@link org.broadinstitute.hellbender.tools.coveragemodel.interfaces.CopyRatioExpectationsCalculator}
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CopyRatioExpectations implements Serializable {
+
+    private static final long serialVersionUID = 7573175667965873997L;
+
+    private final double[] logCopyRatioMeans;
+    private final double[] logCopyRatioVariances;
+    private final int numTargets;
+
+    /**
+     * Public constructor.
+     *
+     * @param logCopyRatioMeans an array of log copy ratio (or copy number) means on a range of targets
+     * @param logCopyRatioVariances an array of log copy ratio (or copy number) variances on a range of targets
+     */
+    public CopyRatioExpectations(@Nonnull final double[] logCopyRatioMeans,
+                                 @Nonnull final double[] logCopyRatioVariances) {
+        this.logCopyRatioMeans = Utils.nonNull(logCopyRatioMeans, "Log copy ratio means array must be non-null");
+        this.logCopyRatioVariances = Utils.nonNull(logCopyRatioVariances, "Log copy ratio variances array must be non-null");
+        Utils.validateArg(logCopyRatioMeans.length == logCopyRatioVariances.length, "The length of log copy ratio means and variances arrays must be equal");
+        numTargets = logCopyRatioMeans.length;
+    }
+
+    /**
+     * Return log copy ratio (or copy number) means
+     *
+     * @return a double array
+     */
+    public double[] getLogCopyRatioMeans() {
+        return logCopyRatioMeans;
```

Comment:
These are big too and I'd rather not spend extra memory. Code safety vs. java's STUPID `final` and lack of real `const` classes/arrays. It is impossible to make objects fully immutable in java anyway without explicit immutable implementations. So I'd rather pick and choose between deep copying and referencing based on the use case, potential for screw-ups, and memory requirements.